pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -634,8 +635,19 @@ def initialize(self):
         # setup spoke state based on data from the Subscription DBus module
         self._update_spoke_state()
 
+        # start the rest of spoke initialization which might take some time
+        # (mainly due to waiting for various initialization threads to finish)
+        # in a separate thread
+        threadMgr.add(AnacondaThread(name=constants.THREAD_SUBSCRIPTION_SPOKE_INIT,
+                                     target=self._initialize))
+
+    def _initialize(self):
         # wait for subscription thread to finish (if any)
         threadMgr.wait(THREAD_SUBSCRIPTION)
+        # also wait for the payload thread, which migh still be processing
+        # a CDROM source, to avoid the Subscription being mandatory by mistake
+        # due to CDN still being default at the time of evaulation
+        threadMgr.wait(THREAD_PAYLOAD)
 
         # update overall state
```

Comment:
I think that the methods bellow that update UI should have the `async_action_wait` decorator.

Reply:
Very good point! They are now running in a thread, which indeed could make GTK go boom.

Reply:
Added the decorates, all seems fine after quick testing. (I have not hit a crash or freeze even before without the decorators, but that could have been luck as most of these GTK threading issues manifest as race conditions.